internal/common/backoff/retry.go

```diff
@@ -30,8 +30,10 @@ type (
 	// Operation to retry
 	Operation func() error
 
-	// IsRetryable handler can be used to exclude certain errors during retry
-	IsRetryable func(error) bool
```

Comment:
we have `IsRetryable` on the server side, but it's just a bool-returning func like this used to be.  Naming-wise though I think either's fine, and I may have settled on `RetryAfter` due to an earlier abandoned attempt 🤔.  I'm game to switch back to `IsRetryable`, seems fine since it returns a bool.

---

As to wait-time: after re-thinking stuff for a bit and half thinking through a few scenarios where it would be defensible to ignore the server's "back off until" time...  

... yeah, we should just trust the server's value in all cases.  Otherwise it can't protect itself from misbehavior, e.g. if a client-side bug leads to hammering the server for some reason.  The lower we can push that logic in code, the less likely it is to be ignored by accident.

And in cases like heartbeats, where there are "meta" timeouts to consider (e.g. the server won't know about the remaining heartbeat limit, because _it can't_, because it's rejecting the request without bothering to do that expensive lookup), we can resolve that later by passing that kind of information in the request and having the server take that into account when rejecting the request.  E.g. if we send a heartbeat with a "we only have 2s left" field, the server can reduce a retry-after down to 1s rather than 5s to give it another chance, effectively prioritizing those heartbeat requests over others.

So since any client-side delay tweaks are reproducible with server-side code and sending more data if necessary, lets just bake in that the server's value is the best one to use.  That way the server can more reliably influence callers when needed.